paimon-core/src/main/java/org/apache/paimon/deletionvectors/DeletionVectorIndexFileWriter.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.deletionvectors;
+
+import org.apache.paimon.fs.FileIO;
+import org.apache.paimon.fs.Path;
+import org.apache.paimon.index.IndexFileMeta;
+import org.apache.paimon.options.MemorySize;
+import org.apache.paimon.table.BucketMode;
+import org.apache.paimon.utils.Pair;
+import org.apache.paimon.utils.PathFactory;
+import org.apache.paimon.utils.Preconditions;
+
+import java.io.Closeable;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.paimon.deletionvectors.DeletionVectorsIndexFile.DELETION_VECTORS_INDEX;
+import static org.apache.paimon.deletionvectors.DeletionVectorsIndexFile.VERSION_ID_V1;
+import static org.apache.paimon.deletionvectors.DeletionVectorsIndexFile.calculateChecksum;
+
+/** Writer for deletion vector index file. */
+public class DeletionVectorIndexFileWriter {
+
+    private final PathFactory indexPathFactory;
+    private final FileIO fileIO;
+    private final boolean isWrittenToMulitFiles;
+    private final long targetSizeInBytes;
+
+    public DeletionVectorIndexFileWriter(
+            FileIO fileIO,
+            PathFactory pathFactory,
+            BucketMode bucketMode,
+            MemorySize targetSizePerIndexFile) {
+        this.indexPathFactory = pathFactory;
+        this.fileIO = fileIO;
+        this.isWrittenToMulitFiles = BucketMode.BUCKET_UNAWARE == bucketMode;
+        this.targetSizeInBytes = targetSizePerIndexFile.getBytes();
+    }
+
+    public List<IndexFileMeta> write(Map<String, DeletionVector> input) throws IOException {
+        if (input.isEmpty()) {
+            return emptyIndexFile();
```

Comment:
If this is possible? Why we need to return a empty Index file?

Reply:
This empty index file is needed to overwrite the previous one when `compact`.
I am going to think it again when to support `compact` operation on deletion vector.

Reply:
here just aligned to the original logic.

Reply:
This may only be to overwrite the old deletion file, as there is no proactive message to delete the deletion file for the primary key table.

But for append table, maybe here no need to generate empty deletion file?